libpod/inspect_data.go

```diff
@@ -0,0 +1,204 @@
+	Name            string           `json:"Name"`
+	RestartCount    int32            `json:"RestartCount"` //not implemented
+	Driver          string           `json:"Driver"`
+	MountLabel      string           `json:"MountLabel"`
+	ProcessLabel    string           `json:"ProcessLabel"`
+	AppArmorProfile string           `json:"AppArmorProfile"`
+	ExecIDs         string           `json:"ExecIDs"` // don't know type yet, not implemented
+	HostConfig      *HostConfig      `json:"HostConfig"`
+	GraphDriver     *driver.Data     `json:"GraphDriver"`
+	SizeRw          int64            `json:"SizeRw,omitempty"`
+	SizeRootFs      int64            `json:"SizeRootFs,omitempty"`
+	Mounts          []specs.Mount    `json:"Mounts"`
+	Config          *CtrConfig       `json:"Config"`
+	NetworkSettings *NetworkSettings `json:"NetworkSettings"` // not implemented
+}
+
+// LogConfig holds the log information for a container
+type LogConfig struct {
+	Type   string            `json:"Type"`   // not implemented
+	Config map[string]string `json:"Config"` //idk type, not implemented
+}
+
+// RestarPolicy holds the container's policy on restart
+type RestartPolicy struct {
+	Name              string `json:"Name"`              // not implemented
+	MaximumRetryCount int    `json:"MaximumRetryCount"` //check type, not implemented
+}
+
+// CtrState represents the state of a container.
+type CtrState struct {
+	OciVersion string    `json:"OciVersion"`
+	Status     string    `json:"Status"`
+	Running    bool      `json:"Running"`
+	Paused     bool      `json:"Paused"`
+	Restarting bool      `json:"Restarting"` // not implemented
+	OOMKilled  bool      `json:"OOMKilled"`
+	Dead       bool      `json:"Dead"`
+	Pid        int       `json:"Pid"`
+	ExitCode   int32     `json:"ExitCode"`
+	Error      string    `json:"Error"` // not implemented
+	StartedAt  time.Time `json:"StartedAt"`
+	FinishedAt time.Time `json:"FinishedAt"`
+}
+
+// HostConfig represents the host configuration for the container
+type HostConfig struct {
+	Binds                []string                    `json:"Binds"`
+	ContainerIDFile      string                      `json:"ContainerIDFile"` //not implemented
+	LogConfig            *LogConfig                  `json:"LogConfig"`       //not implemented
+	NetworkMode          string                      `json:"NetworkMode"`     //not implemented
+	PortBindings         map[string]struct{}         `json:"PortBindings"`    //not implemented
+	RestartPolicy        *RestartPolicy              `json:"RestartPolicy"`   //not implemented
+	AutoRemove           bool                        `json:"AutoRemove"`      //not implemented
+	VolumeDriver         string                      `json:"VolumeDriver"`    //not implemented
+	VolumesFrom          string                      `json:"VolumesFrom"`     // dont know type, not implemented
+	CapAdd               string                      `json:"CapAdd"`          //dont know type, not implemented
```

Comment:
Implemented in kpod run as --cap-add and --cap-drop. Given the amount of stuff we need from kpod run, maybe we do need to store artifacts - maybe not the entire create config, but certainly a subset of it?

Reply:
Also, if libpod can't get these (CapAdd, CapDrop make no sense inside libpod), I would be in favor of removing them from the structs libpod returns (and maybe adding them back in in cmd/kpod?)